### PR TITLE
Create a WidgetItemFactory and use it for better rendering of Jitsi widget change (video conference)

### PR DESCRIPTION
**Pull request by bmarty**
Parity with Element Web

## Before

image

## After

image

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/WidgetItemFactory.kt [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    fun create(event: TimelineEvent,
+               highlight: Boolean,
+               callback: TimelineEventController.Callback?): VectorEpoxyModel<*>? {
+        val widgetContent: WidgetContent = event.root.getClearContent().toModel() ?: return null
+        val previousWidgetContent: WidgetContent? = event.root.resolvedPrevContent().toModel()
+
+        return when (WidgetType.fromString(widgetContent.type ?: previousWidgetContent?.type ?: "")) {
+            WidgetType.Jitsi       -> createJitsiItem(event, callback, widgetContent, previousWidgetContent)
+            WidgetType.TradingView,
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/WidgetItemFactory.kt:60]: use else instead?

**Review by ganfra — APPROVED** [review]
One small remark

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/WidgetItemFactory.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/WidgetItemFactory.kt:60]: Yes, it was to show all the cases we could improve... I will change to `else`

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/WidgetItemFactory.kt]: Done

**Pull request merged**